### ISSUE TITLE
Maintain synapse Feature while using EliteChroma

### ISSUE THREAD

**Issue by fudoido** [poveden/EliteChroma#95]
When connecting motherboard, ram and other rgb light strips/fans in the motherboard, it is all detected in synapse as a single device (when the hardware is compatible with synapse).
By using EliteChroma, it wil give the same animation for all the devices, maintaining its own flow, and not saying is bad, but proposing an alternative on this post.

Synapse has a feature that is Ambient Awareness, Where you can pick a screen region and synapse will use the most "common" color in that region to translate it for all rgb devices connected to motherboard (in my case i am using an msi carbon motherboard with crucial balistix tracer rgb ram, 2 led strips and 2 fans connected to motherboard, all adressable rgb, making it compatible with synapse from manufacturers)
With this in mind, would be possible to keep that feature working when using Elite Chroma, as it will capture the screen environment and provide a better game response (if for example i select the center of the screen and face a star, all my motherboard connected rgb will turn into that star color, if i aim to the empty black space, the rgb will go off and simulate the dark space). This feature also reacts to the supercruise/hyperjump as it will shine as the colors captured ingame when using those functions, as long the selected region is capturing the area where those animations show up in game.
Synapse has for this feature 4 pre-defined areas, a top screen area, a bottom screen area and the 2 lateral areas, just a slim bar around the screen areas so it doesnt capture too many colors. It also has the hability to let you choose the area you wish, where i have chosen the small area where usually the warning messages pop up (so if i get a warning message, my lights will become more red, because that is the more common color in that are when message pops up).
Allow EliteChroma only to take over in specific situation, maybe when fuel scooping, an animation when deploying hardpoints or cargo hold, or when being attacked.
Keyboard and other peripherals would maintain its current status, this suggestion aplies only for any "case" lightning as long hardware is compatible with synapse.

Alternative: Make the above but leave aside the EliteChroma taking over entirely, if it gets too hard to program it in a way that would make EliteChroma take over in specific situations, opposite to permanently on as it is now).

Don't know if this could be possible, and if it can be set as an option in EliteChroma configuration, leaving the option to user to choose from the 2 options (or maybe 3 options)
Option 1 - Full EliteChroma control - app acts like it does now, taking over the entire lightning of the "device" where motherboard, strips/fans and ram are combined
Option 2 - Shared control - EliteChoma reacts like the 1st suggestion, EliteChroma will allow synapse to do most of the animations through synapse "ambient Awareness" feature, and only taking over in specific situations such as fuel scooping or warnings
Option 3 - Synapse Control - EliteChroma will only use peripherals and leave the motherboard "device" totally controlled by synapse, keyboard, mouse, headset and other peripherals would keep EliteChroma current animations and functions, but the combination of lights plugged into motherboard, including ram (in my case the motherboard itself also has some adressable rgb led) detected as a single device by synapse, would be fully controlled by "ambient awareness" feature of synapse.

This suggestion comes due the fact that i enjoy playing with ambient awareness in my pc. as explained above, if i face the dark space, the pc case will be nearly black (leds will turn off or just a faint brightness), when doing supercruise and hyperjump, it catches up blue effects from game and lights up bright blue on my case as well, and if i face a star, or the bright side of a planet, it will light the case as the most visible color that i face, captured by that small region i selected directly from my screen.
It is almost like an ambilight feature, but it captures directly from the screen info, no need for special cameras or other accessories, just a plain screen region selected from synapse software itself.
As a streamer, if i could get this feature from EliteChroma, it would make my streams look more realistic, as i plan to add more strips to my computer on the outside, providing more illumination into my room, and making it feel like the light i am getting on myself, is responding to the lightning i am facing ingame, making it much more immersive, without losing the keyboard key color codes provided by EliteChroma.
I am also considering to buy the razer Chroma RGB hub to connect at my pc, and hopefully this feature could be assigned to that peripheral also (it will be considered a peripheral, since it is not connected directly to motherboard, but through an usb port, and synapse responsive/compatible).

Best regards, thanks for reading the bit big text, hope you understand the suggestion and find a way to make it possible in a easy way.
Marco Matias

**Comment by poveden** [comment]
Wow, that was very thorough! Thank you so much for the ideas!

However, I'm afraid that the Chroma SDK does not work like that. Only one application is in control of _all_ devices, and it's _either_ Chroma Studio (where effects like Ambient Awareness are implemented) _or_ Chroma Connect (where applications like Overwatch or EliteChroma take over).

The [SDK](https://assets.razerzone.com/dev_portal/C%2B%2B/html/en/index.html) includes functionality for starting effects like Chroma Studio, but they are marked as "deprecated and should not be used", and the Ambient Awareness effect is not available anyway.

To complete, setting up a connection with the SDK takes about 1-1.5 seconds, so transitions between device handlers aren't "smooth".

So, to summarize, triggering effects from code is not allowed, and "mixed" scenarios are not possible (not with the current SDK, at least).

Duplicating Ambient Awareness is, of course, technically possible, but it's not an easy undertaking. Considering that I'm developing this tool on my spare time, I'm afraid that some time will pass before I can devote time to this feature.

What I can do is create a new "help requested" issue here in GitHub and perhaps someone will contribute code to the project (or yourself, of course, if you are so inclined :wink:).

What do you think?

**Comment by fudoido** [comment]
thanks for the insight poveden.
there is no chance whatsoever for elitechroma take over only keyboard and leave the other devices under synapse control then?
its a shame, Ambient awareness gives such a more immersion feeling, and if i can add even more led lights to "light up the room" in some scale, it would make players feel like they would be ingame as well.

**Comment by poveden** [comment]
No, I'm afraid there's no chance. I tried that in #78, without success.

I've created #96 to track this feature. Hopefully I will find the time to implement it.

I'm closing this issue now. Thanks again for your feedback!
